app/models/city.rb

```diff
@@ -23,4 +23,8 @@ def display_name
     return "#{parent.name} - #{name}" if parent
     name
   end
+
+  def recent_locations(cutoff_date, limit)
+    EventLocation.joins(:event).where(city: id).where('event_at < ?', cutoff_date).where("bar_name != ''").order('events.event_at desc').take(limit)
```

Comment:
In ruby you can break at any dot, so:

```ruby
EventLocation.
  joins(:event).
  where(city: id).
  where('event_at < ?', cutoff_date).
  where("bar_name != ''").
  order('events.event_at desc').
  take(limit)
```

However I would implement this in a different way.  

At the top of `app/models/city.rb` I would add:

``` ruby
has_many :event_locations  # Sets up a set of association helpers
```

Which will allow you to write:

```ruby
# I would name the method `recent_event_locations` since the association is called `event_locations`
def recent_event_locations(cutoff_at = Time.now, limit = 3)
  event_locations.join(:event).
    where.not(bar_name: ''). # Parenthetically: The fact that we empty bar names is a smell to me. Maybe need an active flag of some sort.
    where('event_at < ?', cutoff_at).
    last(limit)
end
```